GooglePay/src/main/java/com/braintreepayments/api/GooglePayReadinessResult.kt

```diff
@@ -0,0 +1,19 @@
+package com.braintreepayments.api
+
+/**
+ * Result of [GooglePayClient.isReadyToPay]
+ */
+sealed class GooglePayReadinessResult {
+
+    /**
+     * The Google Pay API is supported and set up on this device. Show the Google Pay button for
+     * Google Pay.
+     */
+    object ReadyToPay : GooglePayReadinessResult()
+
+    /**
+     * The Google Pay API is supported or not set up on this device, or there was an [error]
+     * determining readiness.
+     */
+    class NotReadyToPay(val error: Exception?) : GooglePayReadinessResult()
```

Comment:
```suggestion
    class NotReadyToPay(val cause: Exception?) : GooglePayReadinessResult()
```
nit: There is an [Exception constructor](https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/lang/Exception.html#Exception(java.lang.Throwable)) `public Exception(Throwable cause)`. We could borrow the name `cause` to signal to merchants that the exception was produced by an underlying cause.

Reply:
^ The `cause` property also allows us to chain stack traces. It'll be super useful for GitHub inbounds if we leverage that constructor internally when creating `BraintreeException` instances.

Reply:
Updated - do you think we should change to cause for all Failure results or just this one?

Reply:
We should keep it as `Exception` yeah,  `Exception extends Throwable` so we shouldn't have to change any types. There may not always be a `cause` too, and that's fine `cause == null` in that case.

If we do have an exception that signifies the root cause, then a merchant can inspect `braintreeException?.cause` to give us more info. Also, the `toString()` implementation for `Exception` prints out the combined stack trace of the exception chain.

`new BraintreeException("Error with Config", ioExceptionCausedByConfigFetching)` as an example would allow us to chain the exception returned to the merchant by the SDK to the original root cause exception, like in this example the user's WiFi could  be disabled.

Reply:
Ok actually I think the `cause` wording here is confusing - we aren't returning the reason why the user isn't ready to pay (we are just returning NotReadyToPay in those cases). We are only returning an `error` if there was an error determining if the user was ready to pay (in which case it's not really the cause of why they aren't ready to pay, it's that we can't determine if they are ready to pay or not). I'm going to put it back to `error` but I still have a ticket to investigate using cause/throwables

Reply:
Let's also document this design decision to include `cause` where we think it's _applicable_, and some loose guidelines around what "applicable" means somewhere internally.

I can see one of us in the future taking a look at the code and thinking "now why did we do this here but that there?", or even an external dev wondering why we did it 😄

Reply:
I put this back to `error`, which aligns with all other `Failure` results because I think it is more descriptive of what we are actually returning - so no additional design decision doc needed 👍

Reply:
Yeah `cause` is mostly useful for acquiring stack traces from merchants when possible.